e2e/docker-compose.yml

```diff
@@ -112,17 +128,154 @@ services:
+    ports:
+      - 8545:8545
+    working_dir: /tmp/geth
+    volumes:
+      - ./e2e/keystore:/tmp/keystore:ro
+      - ./e2e/passwords.txt:/tmp/passwords.txt:ro
+      - ./e2e/jwt.txt:/tmp/jwt.txt:ro
+      - geth:/tmp/geth
+      - type: tmpfs
+        target: /tmp
+    environment:
+      CHAIN_ID: '900'
+  op-geth-l2:
     build:
-      dockerfile: ./e2e/mocktimism/Dockerfile
-      context: ./..
+      dockerfile: optimism-stack.Dockerfile
+      context: .
+    command:
+      - sh
+      - /tmp/entrypointl2.sh
+    ports:
+      - 8551:8551
+      - 8546:8546
+    volumes:
+      - ./e2e/keystore:/tmp/keystore:ro
+      - ./e2e/passwords.txt:/tmp/passwords.txt:ro
+      - ./jwt.txt:/tmp/jwt.txt:ro
+      - ./entrypointl2.sh:/tmp/entrypointl2.sh
+      - ./genesisl2.sh:/tmp/genesisl2.sh
+      - ./output:/tmp/output
+      - l2configs:/l2configs
+      - ./deploy-config.json:/git/optimism/packages/contracts-bedrock/deploy-config/devnetL1.json
+      - ./deployments:/git/optimism/packages/contracts-bedrock/deployments
+      - ./prestate-proof.json:/git/optimism/op-program/bin/prestate-proof.json
+      - type: tmpfs
+        target: /tmp
+    working_dir: /tmp
+    depends_on: 
+      - op-geth-l1
     environment:
-      MOCKTIMISM_BSS_URL: http://bssd:8081/v1/ws
-      MOCKTIMISM_L2K_RATE_SECONDS: ${HEMI_LOCAL_L2K_RATE_SECONDS}
+      OP_GETH_L1_RPC: 'http://op-geth-l1:8545'
+  op-node:
+    build:
+      dockerfile: optimism-stack.Dockerfile
+      context: .
+    command: 
+      - op-node/bin/op-node
+      - --l2=ws://op-geth-l2:8551
+      - --l2.jwt-secret=/tmp/jwt.txt
+      - --sequencer.enabled
+      - --sequencer.l1-confs=0
+      - --verifier.l1-confs=0
+      - --rollup.config=/l2configs/rollup.json
+      - --rpc.addr=0.0.0.0
+      - --rpc.port=8547
+      - --p2p.disable
+      - --rpc.enable-admin 
+      - --l1=http://op-geth-l1:8545
+      - --l1.rpckind=standard
+      - --l1.trustrpc
+      - --log.level=trace
+      - --l1.trustrpc=true
+      - --l1.http-poll-interval=1s
+    volumes:
+      - l2configs:/l2configs
+      - ./jwt.txt:/tmp/jwt.txt
     depends_on:
-      - bssd
-
+      - op-geth-l1
+      - op-geth-l2
+    environment:
+      OP_NODE_BSS_WS: http://bssd:8081/v1/ws
+    deploy:
+      restart_policy:
+        condition: any
+  op-batcher:
+    build:
+      dockerfile: optimism-stack.Dockerfile
+      context: .
+    command: 
+      - op-batcher/bin/op-batcher
+      - --l2-eth-rpc=http://op-geth-l2:8546
+      - --rollup-rpc=http://op-node:8547
+      - --poll-interval=1s
+      - --sub-safety-margin=1
+      - --num-confirmations=1
+      - --safe-abort-nonce-too-low-count=3
+      - --resubmission-timeout=30s
+      - --rpc.addr=0.0.0.0
+      - --rpc.port=8548 
+      - --rpc.enable-admin
+      - --max-channel-duration=1
+      - --max-pending-tx=1
+      - --l1-eth-rpc=http://op-geth-l1:8545
+      - --private-key=dfe61681b31b12b04f239bc0692965c61ffc79244ed9736ffa1a72d00a23a530
```

Comment:
Can we move private keys and addresses to VARs?

Reply:
ah yes good call 👍